plugins/modules/source_control/git/git_push.py

```diff
@@ -0,0 +1,235 @@
+        type: path
+    user:
+        description:
+            - Git username for https operations.
+        type: str
+    token:
+        description:
+            - Git API token for https operations.
+        type: str
+    branch:
+        description:
+            - Git branch where perform git push.
+        required: True
+        type: str
+    push_option:
+        description:
+            - Git push options. Same as "git --push-option=option".
+        type: str
+    mode:
+        description:
+            - Git operations are performend eithr over ssh, https or local.
+              Same as "git@git..." or "https://user:token@git..." or "git init --bare"
+        choices: ['ssh', 'https', 'local']
+        default: ssh
+        type: str
+    url:
+        description:
+            - Git repo URL.
+        required: True
+        type: str
+requirements:
+    - git>=2.10.0 (the command line tool)
+'''
+
+EXAMPLES = '''
+
+- name: Push changes via HTTPs.
+  community.general.git_push:
+    path: /Users/federicoolivieri/git/git_test_module
+    user: Federico87
+    token: m1Ap!T0k3n!!!
+    branch: master
+    mode: https
+    url: https://gitlab.com/networkAutomation/git_test_module
+
+- name: Push changes via SSH.
+  community.general.git_push:
+    path: /Users/federicoolivieri/git/git_test_module
+    branch: master
+    mode: ssh
+    url: https://gitlab.com/networkAutomation/git_test_module
+
+- name: Push changes on local repo.
+  community.general.git_push:
+    path: /Users/federicoolivieri/git/git_test_module
+    comment: My amazing backup
+    branch: master
+    url: /Users/federicoolivieri/git/local_repo
+'''
+
+RETURN = '''
+output:
+    description: list of git cli commands stdout
+    type: list
+    returned: always
+    sample: [
+        "To https://gitlab.com/networkAutomation/git_test_module.git\n   372db19..99830f4  master -> master\n"
+    ]
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+
+
+def git_push(module):
+
+    commands = list()
+
+    path = module.params.get('path')
+    url = module.params.get('url')
+    user = module.params.get('user')
+    token = module.params.get('token')
+    branch = module.params.get('branch')
+    push_option = module.params.get('push_option')
+    mode = module.params.get('mode')
+
+    def https(path, user, token, url, branch, push_option):
+        if url.startswith('https://'):
+            remote_add = 'git -C {path} remote set-url origin https://{user}:{token}@{url}'.format(
+                path=path,
+                url=url[8:],
+                user=user,
+                token=token,
+            )
+            cmd = 'git -C {path} push origin {branch}'.format(
+                path=path,
+                branch=branch,
+            )
+
+        if push_option:
+            index = cmd.find('origin')
+            return [remote_add, cmd[:index] + '--push-option={option} '.format(option=push_option) + cmd[index:]]
```

Comment:
This will break if `url` does not starts with `https://`, since then `remote_add` has not been defined.

Reply:
That would be just meter to indent that block under `if url.startswith('https://'):` Would that work for you?

```
    def https(path, user, token, url, branch, push_option):
        if url.startswith('https://'):
            remote_add = 'git -C {path} remote set-url origin https://{user}:{token}@{url}'.format(
                path=path,
                url=url[8:],
                user=user,
                token=token,
            )
            cmd = 'git -C {path} push origin {branch}'.format(
                path=path,
                branch=branch,
            )

            if push_option:
                index = cmd.find('origin')
                return [remote_add, cmd[:index] + '--push-option={option} '.format(option=push_option) + cmd[index:]]

            if not push_option:
                return [remote_add, cmd]
```

Reply:
You should rather raise an error when `url` does not start with `https://`.